sratatata

mateuszde opened the pull request:
add style.css
add app.js
add index.html

mateuszde committed:
add index.html
add style.css
add app.js

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,123 @@
+
+}
+
+const toDo = new ToDoList();
```
Rada na przyszłość - pamiętaj, żeby ZAWSZE zostawiać pustą linię na końcu pliku. To się tyczy każdego, JSa, HTMLa czy CSSów. 

Żeby zbytnio nie tłumaczyć chodzi o to, że przy minifikacji czasami pliki bez tej koncowej pustej linii się po prostu 'psują'. W skrócie.

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,123 @@
+class ToDoList {
+    constructor() {
+        this.toDoTasks = [
```
Podejrzewam, że zostawiłeś to sobie, żeby pamiętać jak wyglądają obiekty zadań, polecam do tego JSdocsa lub po prostu nauczyć się TypeScripta (szczerze mówiąc do tego mocno nakłaniam, mega działa na wyobraźnię przy rozmowach kwalifikacyjnych), ktory na Tobie wymusi użycie interfejsu ktory jest mega pomocny w czymś takim.

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,123 @@
+        this.inputAddTask = document.getElementById('addTaskInput');
+        this.taskSection = document.querySelector('.toDoTasks');
+
+        this.render();
```
Zamiast wywoływać render w constructorze, lepiej jest to zrobić na obiekcie `toDo` (`toDo.render()` na samym koncu pliku). Wywoływanie funkcji w constructorze jest ogólnie niezalecane.

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,123 @@
+
+        document.getElementById('addTaskBtn').addEventListener('click', this.addTask);
+        this.inputAddTask = document.getElementById('addTaskInput');
+        this.taskSection = document.querySelector('.toDoTasks');
```
To jest akurat moja dygresja, aniżeli jakaś wielka konwencja, ale ja mam w zwyczaju używać id'ków (i wtedy `.getElementById` jak to zrobiłeś w linii 22) do elementów które w domyśle występują raz. W tym konkretnym przypadku nie ma to większego znaczenia, ale przy większej aplikacji jak się kiedyś jebniesz i zapomnisz o istnieniu drugiej sekcji o podobnej klasie.. No to wtedy jest klopsik :)

vonsky104 replied:
A uwierz mi, przy dużych appkach zapomnieć o cssowej klasie ktora miała występować raz a już nie występuje zdarza się baaaardzo często :)

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,123 @@
+
+    }
+
+    render = (taskList = this.toDoTasks) => {
```
Ustawianie domyślnej wartości argumentu funkcji, która przy wywoływaniu nie ma żadnego przekazania elementu nie ma zbytnio sensu. Coś takiego stosujesz wtedy gdy dopuszczasz sytuację, że przekazany argument np jest `undefined`. W tym przypadku `render` wywołujesz w constructorze nieprzekazując nic. 
Także już lepiej zrobić wewnątrz `render`: `const taskList = this.toDoTasks` niż ustawiać domyślną wartość. 
Druga opcja to po prostu przekazanie do funkcji `render` listy i wtedy wywalenie domyślnej wartości.

vonsky104 reviewed (changes requested):
Request Changes

mateuszde committed:
Update readme.md

mateuszde committed:
add new line at the end of file - style.css, app.js

mateuszde committed:
Change app.js

- add createTask function
- change addTask function
- change removeTask function
- change changeStatus function
- change render function
- optimatization rendering

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    render = () => {
+        this.clearView();
```
W jakiej sytuacji to się właściwie wykonuje? Render działa tylko na starcie, więc właściwie nie ma czego usuwać.

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,121 @@
+        }
+    }
+
+    changeStatus = (e, taskList = this.toDoTasks) => {
```
Nie przekazujesz nic poza `e` do `changeStatus` więc podobny case z domyślną wartością argumentu jak wcześniej w renderze. Niepotrzebny.

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,121 @@
+        this.createTask(taskItem.id, taskItem.task, taskItem.completed);
+    }
+
+    removeTask = (e, taskList = this.toDoTasks) => {
```
Nie przekazujesz nic poza `e` do `removeTask` więc podobny case z domyślną wartością argumentu jak wcześniej w renderze. Niepotrzebny.

vonsky104 commented on app.js:
```diff
@@ -0,0 +1,121 @@
+        deleteButton.addEventListener('click', this.removeTask);
+    }
+
+    addTask = (e, task = this.inputAddTask.value, id = Date.now()) => {
```
Identyczna sytuacja jak z `removeTask` i `changeStatus`, domyślne wartości argumentów zawsze będą takie jakie im tu przypisujesz.  Zadeklaruj `task` i `id` wewnątrz funkcji.

vonsky104 reviewed (commented):
Jeszcze to

mateuszde committed:
change app.js

- delete unnecessarily passed arguments in such functions like: render, changeStatus, removeTask, addTask
- add event listener on 'Enter' to add addTask
- add autofocusing on input after clicking 'Dodaj' button
